tools/dockerise/__init__.py

```diff
@@ -40,20 +40,26 @@ def build_platform(platform):
             os.chmod(p, current & ~(stat.S_IWGRP | stat.S_IWOTH))
 
     # Pull the latest version from dockerhub
-    err = pty.spawn(["docker", "pull", remote_tag])
+    build_env = os.environ
+    build_env["DOCKER_BUILDKIT"] = "1"
+    err = pty.spawn(["docker", "pull", remote_tag], env=build_env)
     if err != 0:
         cprint("Docker pull returned exit code {}".format(err), "red", attrs=["bold"])
         exit(err)
 
-    build_env = os.environ
-    build_env["DOCKER_BUILDKIT"] = "1"
     old_cwd = os.getcwd()
     os.chdir(dockerdir)
+    # The following link suggets that a comma-separated list should be used for caching from multiple images
+    # and that the order is important. Since the local_tag may contain local modifications we should specify
+    # it first
+    # https://github.com/moby/moby/issues/34715#issuecomment-425933774
     err = pty.spawn(
         [
             "docker",
             "build",
             ".",
+            "--cache-from",
+            "{},{}".format(local_tag, remote_tag),
```

Comment:
```suggestion
            ','.join([local_tag, remote_tag]),
```

Reply:
Thats why `local_tag` is specified first so it will take its cache from that image (assuming that `local_tag` is your locally modified image and `remote_tag` is master)